core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreMessageHandler.java

```diff
@@ -15,28 +15,62 @@
 
 package io.confluent.kafka.schemaregistry.storage;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import io.confluent.kafka.schemaregistry.avro.AvroSchema;
 import io.confluent.kafka.schemaregistry.id.IdGenerator;
 import io.confluent.kafka.schemaregistry.metrics.MetricsContainer;
 import io.confluent.kafka.schemaregistry.metrics.SchemaRegistryMetric;
+import io.confluent.kafka.schemaregistry.rest.SchemaRegistryConfig;
 import io.confluent.kafka.schemaregistry.storage.exceptions.StoreException;
 import org.apache.kafka.common.TopicPartition;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.Arrays;
+
 public class KafkaStoreMessageHandler implements SchemaUpdateHandler {
 
   private static final Logger log = LoggerFactory.getLogger(KafkaStoreMessageHandler.class);
   private final KafkaSchemaRegistry schemaRegistry;
   private final LookupCache<SchemaRegistryKey, SchemaRegistryValue> lookupCache;
   private IdGenerator idGenerator;
+  private File backupFile;
 
   public KafkaStoreMessageHandler(KafkaSchemaRegistry schemaRegistry,
                                   LookupCache<SchemaRegistryKey, SchemaRegistryValue> lookupCache,
-                                  IdGenerator idGenerator) {
+                                  IdGenerator idGenerator,
+                                  String backupsDir,
+                                  String backupFilePrefix) {
     this.schemaRegistry = schemaRegistry;
     this.lookupCache = lookupCache;
     this.idGenerator = idGenerator;
+    this.backupFile = new File(backupsDir + "/" + backupFilePrefix + ".txt");
+    if (schemaRegistry.config().getBoolean(SchemaRegistryConfig.WRITE_BACKUPS)) {
+      try {
+        File directory = new File(backupsDir);
+        if (!directory.exists()) {
+          if (!directory.mkdir()) {
+            log.error("failed to create directory");
+          } else {
+            log.info("created new directory: " + directory.getAbsolutePath());
+          }
+        }
+        if (!backupFile.exists()) {
+          if (!backupFile.createNewFile()) {
+            log.error("failed to create backup file");
+          } else {
+            log.info("created new file: " + backupFile.getAbsolutePath());
+          }
+        }
+      } catch (IOException e) {
+        log.error("exception when trying to create backup file");
```

Comment:
add the filename which did not get created. Add a metric as well stating backup file creation failed.